File: backend/src/main/java/com/serenade/backend/domain/track/dto/TrackResponse.java
package com.serenade.backend.domain.track.dto;

import com.serenade.backend.domain.track.Genre;
import com.serenade.backend.domain.track.Track;

import java.time.Instant;
import java.util.UUID;

public record TrackResponse(
        UUID id,
        String title,
        String artist,
        String album,
        Genre genre,
        Long durationMs,
        String artworkUrl,
        String streamUrl,
        Instant updatedAt
) {
    public static TrackResponse from(Track t) {
        return new TrackResponse(
                t.getId(), t.getTitle(), t.getArtist(), t.getAlbum(),
                t.getGenre(), t.getDurationMs(), t.getArtworkUrl(), t.getStreamUrl(),
                t.getUpdatedAt()
        );
    }
}

File: app/app/src/main/java/com/serenade/app/feature/track/data/remote/dto/TrackResponse.kt
package com.serenade.app.feature.track.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class TrackResponse(
    val id: String,
    val title: String,
    val artist: String,
    val album: String? = null,
    val genre: String,
    val durationMs: Long? = null,
    val artworkUrl: String? = null,
    val streamUrl: String? = null,
    val updatedAt: String? = null,
)
